source/byte_buf.c

```diff
@@ -711,12 +726,17 @@ int aws_byte_cursor_compare_lookup(
     }
 
     if (lhs_curr < lhs_end) {
+        AWS_POSTCONDITION(aws_byte_cursor_is_valid(lhs));
```

Comment:
Do we need to write these postconditions three times? It seems that we assert them no matter what we're returning...suggest writing the postconditions just above `if (lhs_curr < lhs_end)`.

Reply:
Nice catch!